hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/FSWALEntry.java

```diff
@@ -56,19 +58,24 @@
   private final transient boolean inMemstore;
   private final transient RegionInfo regionInfo;
   private final transient Set<byte[]> familyNames;
+  private final transient Optional<ServerCall<?>> rpcCall;
 
-  FSWALEntry(final long txid, final WALKeyImpl key, final WALEdit edit,
-      final RegionInfo regionInfo, final boolean inMemstore) {
+  FSWALEntry(final long txid, final WALKeyImpl key, final WALEdit edit, final RegionInfo regionInfo,
+    final boolean inMemstore, ServerCall<?> rpcCall) {
     super(key, edit);
     this.inMemstore = inMemstore;
     this.regionInfo = regionInfo;
     this.txid = txid;
     if (inMemstore) {
       // construct familyNames here to reduce the work of log sinker.
-      Set<byte []> families = edit.getFamilies();
-      this.familyNames = families != null? families: collectFamilies(edit.getCells());
+      Set<byte[]> families = edit.getFamilies();
+      this.familyNames = families != null ? families : collectFamilies(edit.getCells());
```

Comment:
Unrelated issue:   I see the collectFamilies internal ,  lots of Stream & lamda code there... That would be a performance issue...

Reply:
The above issue can be addressed in this patch I think,  seems an easy fix.